src/tool/xmeta/cpp/xlang_model_walker.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include "xmeta_models.h"
+#include "xlang_model_listener.h"
+
+namespace xlang::xmeta
+{
+    class xlang_model_walker
+    {
+    public:
+        xlang_model_walker(std::map<std::string_view, std::shared_ptr<namespace_model>, std::less<>> const& namespaces, xlang_model_listener & listener)
+            : m_namespaces(namespaces), m_listener(listener)
+        { };
+
+        void register_listener(xlang_model_listener & listener);
+        void walk();
+        
+        xlang_model_walker() = delete;
+
+    private:
+        xlang_model_listener & m_listener;
```

Comment:
We can make this a `std::vector<std::shared_ptr<xlang_model_listener>> m_listeners;` to allow multiple listeners to be attached.

Reply:
Actually I don't really see the need for it right now so I will just leave it as is.